app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -350,13 +353,55 @@ def populate_hardware_hash_with_series_attributes(hardware_hash, instance, serie
         hardware_hash[:memory_mb]       = series[:memory] / 1.megabyte
         hardware_hash[:disk_capacity]   = series[:root_disk_size] + series[:swap_disk_size]
 
-        os_disk = instance.properties.storage_profile.os_disk
-        sz      = series[:root_disk_size]
-        vhd_loc = os_disk.try(:vhd).try(:uri) || os_disk.try(:managed_disk).try(:id)
+        add_instance_disk(hardware_hash[:disks], instance)
+
+        # No data available on swap disk? Called temp or resource disk.
+      end
+
+      # Redefine the inherited method for our purposes
+      def add_instance_disk(array, instance, disk = nil)
+        disk ||= instance.properties.storage_profile.os_disk
+
```

Comment:
can you pass the OS Disk in as a parameter so its the same as you do for data disks? This line is OS disk specific yet the method is used for all disks.

Reply:
Ok, done.